libraries/AP_HAL_Linux/Util.cpp

```diff
@@ -145,21 +146,22 @@ int Util::read_file(const char *path, const char *fmt, ...)
 {
     errno = 0;
 
-    FILE *file = ::fopen(path, "re");
-    if (!file)
+    FILE *file = fopen(path, "re");
+    if (!file) {
         return -errno;
+    }
 
     va_list args;
     va_start(args, fmt);
 
-    int ret = ::vfscanf(file, fmt, args);
-    int errno_bkp = errno;
-    ::fclose(file);
+    int ret = vfscanf(file, fmt, args);
+    fclose(file);
 
     va_end(args);
 
-    if (ret < 1)
-        return -errno_bkp;
+    if (ret < 1) {
+        return -errno;
```

Comment:
The `ret` variable is related to the call to `vfscanf()` and you're returning `errno` which might be changed by `fclose()`. The `errno_bkp` variable shouldn't be removed.